### DISCUSSION TITLE
How to give permissions to specific players in creating claims to a specific world?

### DISCUSSION THREAD

**Discussion by eliyash31** [GriefPrevention/GriefPrevention#1012] [Q&A] [answered]
Hello, I can't seem to find in the config and in the permissions page about giving permissions to specific players in a specific world. Like for example I want to give permissions to higher rank players to be able to claim in end and nether. please help thanks!

**Comment by bigpresh** [comment]
Pretty sure there's no way to do that, at least at present.

The checks for "has a permission that allows creating claims" is done by the server based on the `plugin.yml`.

It would be possible to replace those overall checks with checks inside of GP - e.g. that you have either `griefprevention.claims` (to let you claim in all claim-enabled worlds) or have `griefprevention.$world.claims` to allow you to claim in that specific world), but I don't know whether that's a feature which would be widely useful enough to go into GP core, or better handled by an addon.

(An addon to do this could catch claim creation attempts and perform additional checks like that easily enough.)

**Comment by eliyash31** [comment]
Okay thank you so much, I would just warn the players that don't have permissions to not try to claim in those world or else they will get a punishment

**Comment by Muspah** [comment, marked as answer]
We used to implement this by requiring permission to create claims, and then only handing out the permission to do so in specific worlds to specific users/groups. zPermissions and LuckPerms were able to handle that.

**Comment by bigpresh** [comment]
Ah yes, of course, using a permissions plugin which gives you permissions based on the world you're in seems like it should work just fine - good point @Muspah !

**Comment by RoboMWM** [comment]
Yup, negating perms for the lower ranks should work, depends how you have inheritance and priorities setup. @eliyash31

**Comment by eliyash31** [comment]
Okay thank you so much! I will ask help for the devs in LuckPerms on how to implement this idea. Thank you for the help :)

**Comment by Muspah** [comment]
If you're having problems implementing this, I could take a look in our setup. Shouldn't be that hard iirc.
